modules/oe_content_call_proposals/oe_content_call_proposals.module

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/**
+ * @file
+ * Primary module hooks for OpenEuropa Call For Proposals content module.
+ */
+
+declare(strict_types = 1);
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\oe_content_call_proposals\CallForProposalsNodeWrapperInterface;
+
+/**
+ * Implements hook_form_BASE_FORM_ID_alter for the node form.
+ */
+function oe_content_call_proposals_form_node_form_alter(array &$form, FormStateInterface $form_state) {
+  /** @var \Drupal\node\NodeInterface $node */
+  $node = $form_state->getBuildInfo()['callback_object']->getEntity();
+  if ($node->bundle() !== 'oe_call_proposals') {
+    return;
+  }
+
+  $form['oe_call_proposals_deadline']['widget']['#states'] = [
+    'invisible' => [
+      'select[name="oe_call_proposals_model"]' => [
+        'value' => CallForProposalsNodeWrapperInterface::MODEL_PERMANENT,
+      ],
+    ],
+  ];
+}
+
+/**
+ * Implements hook_entity_type_alter().
+ */
+function oe_content_call_proposals_entity_type_alter(array &$entity_types) {
+  $entity_types['node']->addConstraint('CallProposalsJournalTitleLength');
+  $entity_types['node']->addConstraint('CallProposalsDeadlineDate');
+}
+
+/**
+ * Returns the Model's allowed values.
+ */
+function oe_content_call_proposals_model_allowed_values() {
```

Comment:
since this is a custom function and not part of the drupal api lets implement return type hinting in accordance with the strict types
```suggestion
function oe_content_call_proposals_model_allowed_values():void {
```

Reply:
This is fixed.